GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -865,7 +867,38 @@ protected override void OnMouseWheel(MouseEventArgs e)
             }
             else
             {
-                base.OnMouseWheel(e);
+                if (_lastMouseWheel.ElapsedMilliseconds > 1500)
```

Comment:
Why 1500, why is this needed at all?
Why is wheelDeltaPerRow  120? (the comment that 120 corresponded to one row may be needed)

Reply:
> Why is wheelDeltaPerRow 120?

This seems to be a historic value and the common definition: [https://learn.microsoft.com/.../wm-mousewheel](https://learn.microsoft.com/en-us/windows/win32/inputdev/wm-mousewheel)

I just found out that this value can be accessed via `SystemInformation.MouseWheelScrollDelta` which is just a getter for the internal constant `WHEEL_DELTA = 120`: [https://github.com/dotnet/.../SystemInformation.cs#L884](https://github.com/dotnet/winforms/blob/d892c6c3186f26e763a9f54f80a1379c845e1c4d/src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs#L884)
I will make a change to use `SystemInformation.MouseWheelScrollDelta`. Should I still add comment, and what information should it include?

> Why is the reset timer needed at all?

I stumbled upon this when I tested "smooth scrolling" and changed back to notched scrolling. While a notched wheel produces delta values that are multiples of 120, smooth scrolling produces arbitrary values and therefore leaves behind unconsumed total delta (since it's consumed in multiples of 120). Now assume the total delta is, say, -100 und you rotate the wheel by one notch in the positive direction, then nothing will happen, because the total delta is now at +20, but the delta needed for scrolling one row is 120. Whenever you change direction, the first notch will do nothing. I can imagine myself experiencing this when using a laptop with a mouse attached and switching between touchpad (= smooth scrolling) and mouse randomly.
If you use only a touchpad or a free-spinning mouse, you might also experience a short "lag" or "jump" at the beginning of a scroll for the same reasons.

There might be more intricate ways to reset the total wheel delta (all needing a timer somehow, I guess), but a single timer is presumably good enough, and simple of course.

> Why 1500?

It's a recommendation I found here: https://bugzilla.mozilla.org/show_bug.cgi?id=605648#c90

Reply:
Thanks for the explanation. The use of the Suggested comments - did I understand?

```
// Mouse wheel idle: Reset unused delta
                if (_lastMouseWheel.ElapsedMilliseconds > 1500)

...
// All scroll movement may not be used, save to next event
_mouseWheelDeltaRemainder = totalWheelDelta % wheelDeltaPerRow;
```

Move the following to the beginning of the else (or even function, also for horizontal scroll)
```
int maxRowIndex = _revisionGraph.Count - 1;
                    if (maxRowIndex < 0)
                    {
                        return;
                    }
```

Reply:
Yes, correct. I added comments per your suggestion, thanks.

Reply:
> I stumbled upon this when I tested "smooth scrolling" and changed back to notched scrolling. While a notched wheel produces delta values that are multiples of 120, smooth scrolling produces arbitrary values and therefore leaves behind unconsumed total delta (since it's consumed in multiples of 120). Now assume the total delta is, say, -100 und you rotate the wheel by one notch in the positive direction, then nothing will happen, because the total delta is now at +20, but the delta needed for scrolling one row is 120. Whenever you change direction, the first notch will do nothing. I can imagine myself experiencing this when using a laptop with a mouse attached and switching between touchpad (= smooth scrolling) and mouse randomly. If you use only a touchpad or a free-spinning mouse, you might also experience a short "lag" or "jump" at the beginning of a scroll for the same reasons.
> 
> There might be more intricate ways to reset the total wheel delta (all needing a timer somehow, I guess), but a single timer is presumably good enough, and simple of course.

This looks like a very useful context, and I suggest adding it as a comment to the code.